### DISCUSSION TITLE
Run a task for all dependencies of some target?

### DISCUSSION THREAD

**Discussion by NullVoxPopuli** [vercel/turborepo#987] [Help] [answered]
note that this is related to: https://github.com/vercel/turborepo/issues/417
An alternative solution I thought of to #417 is : 
 - build all dependencies for the target project
 - then start the project
 
 e.g.: `"start": "<build-my-deps> && start-dev-server"`
 
 so, what _this_ question is about is how to craft a `turbo` command that would fit in the `<build-my-deps>` part of the above.
 
 why is this important? 
 in a big monorepo with many projects, you may have 100+ libraries and 50 apps -- if one app you're about to work in only pulls in 10 of those libraries, you don't need to also build the other 90 libraries (or other 49 apps)
 
now, I _could_ hack this by adding a no-op package.json#script to to each target project -- however this means that, because turborepo (currently), does not supporting defining a task/command graph (for example start -> runs build in all dependents (what #417 is about)), I would not be able to do this development strategy for every layer of development (developing the app could work with this strategy, but then I couldn't also develop any of the packages with the same command -- this doesn't work recursively is what I'm saying).

so, the only thing that I think could work, (short of the solution to #417), is to somehow get a list of dependencies, and run a command on each of those.

For example (pseudo CLI): `turbo run build --in-dependencies-of some-app`
or maybe, `turbo run-in-each build $(turbo dependenciesof some-app)`

is this possible with today's commands / args?

I saw recently that `--filter` was implemented, but I'm not quite sure if it helps in this case, or if it's just an important piece towards getting to the above describe behavior

Much thanks! <3

**Comment by elado** [comment, marked as answer]
Related: https://github.com/vercel/turborepo/issues/965
